sink/sqs.go

```diff
@@ -0,0 +1,164 @@
+package sink
+
+import (
+	"strconv"
+	"time"
+
+	"os"
+
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	log "github.com/sirupsen/logrus"
+)
+
+// SQS ...
+type SQSSink struct {
+	session   *session.Session
+	sqs       *sqs.SQS
+	queueName string
+	groupId   string
+	stopCh    chan interface{}
+	putCh     chan []byte
+	batchCh   chan [][]byte
+}
+
+// NNewSQS ...
+func NewSQS(groupId string) (*SQSSink, error) {
+	queueName := os.Getenv("SINK_SQS_QUEUE_URL")
+	if queueName == "" {
+		return nil, fmt.Errorf("[sink/sqs] Missing SINK_SQS_QUEUE_URL")
+	}
+
+	sess := session.Must(session.NewSession())
+	svc := sqs.New(sess)
+
+	return &SQSSink{
+		session:   sess,
+		sqs:       svc,
+		queueName: queueName,
+		groupId:   groupId,
+		stopCh:    make(chan interface{}),
+		putCh:     make(chan []byte, 10000),
+		batchCh:   make(chan [][]byte, 100),
+	}, nil
+}
+
+// Start ...
+func (s *SQSSink) Start() error {
+	// Stop chan for all tasks to depend on
+	s.stopCh = make(chan interface{})
+
+	go s.batch()
+	go s.write()
+
+	// wait forever for a stop signal to happen
+	for {
+		select {
+		case <-s.stopCh:
+			break
+		}
+		break
+	}
+
+	return nil
+}
+
+// Stop ...
+func (s *SQSSink) Stop() {
+	log.Infof("[sink/sqs] ensure writer queue is empty (%d messages left)", len(s.putCh))
+
+	for len(s.putCh) > 0 {
+		log.Infof("[sink/sqs] Waiting for queue to drain - (%d messages left)", len(s.putCh))
+		time.Sleep(1 * time.Second)
+	}
+
+	close(s.stopCh)
+}
+
+// Put ..
+func (s *SQSSink) Put(data []byte) error {
+	s.putCh <- data
+
+	log.Infof("[sink/sqs] (%d messages left)", len(s.putCh))
```

Comment:
can we remove this line? it will be quite spammy and the other sinks don't log here